packages/ui-kit/src/components/Logger/Log.types.ts

```diff
@@ -0,0 +1,14 @@
+export enum LogLevels {
+  Error = 'error',
+  Warn = 'warn',
+  Info = 'info',
+  Debug = 'debug'
+}
+
+export type SupportedConsole = Pick<Console, LogLevels>
+
+export type UseLogConfig = {
+  environments?: string[]
+  logLevel?: LogLevels
+  console?: Console
+}
```

Comment:
Why not have a single type we return,

```ts
export interface Log {
  /** The current log level.*/
  level: LogLevel

  /** Log at error-level, if enabled. */
  error: Console["error"]

  /** Log at warn-level, if enabled. */
  warn: Console["warn"]

  /** Log at info-level, if enabled. */
  info: Console["info"]

  /** Log at debug-level, if enabled. */
  debug: Console["debug"]
}
```

and do

```ts
const log = useLog()
log.info(`the level is ${log.level}`)
```

do we actually need the extra wrapper?

```ts
const { log, logLevel } = useLog()
log.info(`the level is ${logLevel}`)
```